third_party/torch-mlir-dialects/lib/Dialect/TMTensor/IR/TMTensorOps.cpp

```diff
@@ -438,6 +438,166 @@ LogicalResult ScatterOp::generateScalarImplementation(OpBuilder &b,
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// SortOp
+//===----------------------------------------------------------------------===//
+
+LogicalResult SortOp::verify() {
+  Operation *op = getOperation();
+  if (getNumInputs()) {
+    return op->emitOpError("does not expect to take any inputs");
+  }
+  if (getNumOutputs() == 0) {
+    return op->emitOpError("expected at least one `outs` operand");
+  }
+
+  Block &block = getRegion().front();
+  size_t numOutputs = getNumOutputs();
+  if (block.getNumArguments() != 2 * numOutputs) {
+    return op->emitOpError("region block should have ")
+           << 2 * numOutputs << " arguments";
+  }
+
+  int64_t rank = getOperandRank();
+  int sortDim = getDimension();
+  if (sortDim < 0 || sortDim >= rank) {
+    return op->emitOpError("dimension must be within (0, ") << rank << "]";
+  }
+
+  ArrayRef<int64_t> shape = getOperandShape();
+  for (auto indexedOperand : llvm::enumerate(getOutputs())) {
+    int index = indexedOperand.index();
+    auto operandType = getOperandType(index);
+    if (operandType.getRank() != rank) {
+      return op->emitOpError("expected operand ")
+             << index << " to be rank " << rank << ", same as other operands";
+    }
+    if (operandType.getShape() != shape) {
+      return op->emitOpError("expected operand ")
+             << index << " to have same shape as other operands";
+    }
+    Type elemType = operandType.getElementType();
+    for (int i : {2 * index, 2 * index + 1}) {
+      Type argType = block.getArgument(i).getType();
+      if (argType != elemType) {
+        return op->emitOpError("region block argument #")
+               << i << " should be of type " << elemType << " but got "
+               << argType;
+      }
+    }
+  }
+
+  auto yieldOp = cast<YieldOp>(block.getTerminator());
+  if (yieldOp.getNumOperands() != 1) {
+    return op->emitOpError("should yield exactly one operand");
+  }
+  auto ty = yieldOp.getOperand(0).getType().dyn_cast<IntegerType>();
```

Comment:
Nit: new MLIR style is dyn_cast<A>(b) vs b.dyn_cast<A>() that was previously required.